### PR TITLE
Initial pull request

### PR DESCRIPTION
**Pull request by brucehoff**

### CODE REVIEW

File: src/putty/cmdline.c [hunk 1]
```diff
@@ -0,0 +1,575 @@
...
+    }
+    saves[pri].params[saves[pri].nsaved].p = p;
+    saves[pri].params[saves[pri].nsaved].value = value;
+    saves[pri].nsaved++;
+}
+
+static char *cmdline_password = NULL;
+
+void set_user_password(char *pass) {
+	cmdline_password = snewn(strlen(pass)+1, char);
```
Comment (marcel-blonk) [inline src/putty/cmdline.c:62]: check memory usage. Also, free if password already allocated

File: src/putty/psftp.c [hunk 2]
```diff
@@ -0,0 +1,3157 @@
...
+     */
+    assert(len == 0);
+}
+
+/*
+ * In psftp, all agent requests should be synchronous, so this is a
+ * never-called stub.
+ */
+void agent_schedule_callback(void (*callback)(void *, void *, int),
+			     void *callba
```
Comment (marcel-blonk) [inline src/putty/psftp.c:2525]: Free or use stack variable

Comment (marcel-blonk) [inline src/putty/psftp.c:2525]: Same

Comment (marcel-blonk) [inline src/putty/psftp.c:2525]: Same

File: src/putty/cmdline.c (hunk 1 above)
Comment (brucehoff) [inline src/putty/cmdline.c:62]: I checked:  My "clean up" function, rssh_clean_up() calls cmdline_cleanup(), which clears and deallocates "cmdline_password".

**Pull request merged**